src/store/login/saga.ts

```diff
@@ -77,6 +77,14 @@ export function* web3Login(action) {
   }
 }
 
+/**
+ * Switches login stage. Used for switching between login options when logging in.
+ */
+export function* switchLoginStage(action) {
+  yield put(setErrors([]));
+  yield put(setStage(action.payload));
+}
+
```

Comment:
Wasn't sure if this was the right way to do things - I think this should probably just be held in state in `LoginContainer`, but curious to see if this was correct

Reply:
> I think this should probably just be held in state in LoginContainer
No, this is the correct place. Keeping user flow logic in redux/sagas allows a greater level of control and flexibility.

There's a couple of options here, I think:
1. Add the preventative validation. But, what would you do? Just fail silently and not switch probably. Simple but doesn't really help the developer.
2. Have 2 separate functions that only do the specific thing: `switchToEmailLogin`/`switchToWeb3Login`. This has the benefit of definitely only allowing the 2 things we want. I think I prefer this option. We're only going to have 2 options so there's no concern about an explosion of login types that we have to worry about and it's very explicit what's permitted.

Reply:
For _1_ I would ideally like to set the types such that you can only provide `Email` or `Web3` as params. This only really helps for DX though, so I think _2_ is the better solution!